SU2_CFD/include/limiters/computeLimiters_impl.hpp

```diff
@@ -0,0 +1,260 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+/*!
+ * \brief General limiter computation for methods based on one limiter
+ *        value per point (as opposed to one per edge) and per variable.
+ * \note This implementation can be used to derive most common methods
+ *       by specializing the limiter functions (e.g. Venkatakrishnan)
+ *       and the geometric modifications (e.g. sharp edges), this is done
+ *       via specialization of "CLimiterDetails" (all its methods). Then,
+ *       a call to this function with the specialized "LimiterKind" should
+ *       be added to the body of "computeLimiters()".
+ *       See also the notes in computeGradientsGreenGauss.hpp
+ *
+ * Arguments:
+ * \param[in] solver - Optional, solver associated with the field (used only for MPI).
+ * \param[in] kindMpiComm - Type of MPI communication required.
+ * \param[in] kindPeriodicComm1 - Type of periodic comm. to determine min/max.
+ * \param[in] kindPeriodicComm2 - Type of periodic comm. to adjust limiters.
+ * \param[in] geometry - Geometric grid properties.
+ * \param[in] config - Configuration of the problem.
+ * \param[in] varBegin - First variable index for which to compute limiters.
+ * \param[in] varEnd - End of computation range (nVar = end-begin).
+ * \param[in] field - Variable field.
+ * \param[in] gradient - Gradient of the field.
+ * \param[out] fieldMin - Minimum field values over direct neighbors of each point.
+ * \param[out] fieldMax - As above but maximum values.
+ * \param[out] limiter - Reconstruction limiter for the field.
+ *
+ * Template parameters:
+ * \param FieldType - Generic object with operator (iPoint,iVar)
+ * \param GradientType - Generic object with operator (iPoint,iVar,iDim)
+ * \param LimiterKind - Used to instantiate the right details class.
+ */
+template<class FieldType, class GradientType, ENUM_LIMITER LimiterKind>
+void computeLimiters_impl(CSolver* solver,
+                          MPI_QUANTITIES kindMpiComm,
+                          PERIODIC_QUANTITIES kindPeriodicComm1,
+                          PERIODIC_QUANTITIES kindPeriodicComm2,
+                          CGeometry& geometry,
+                          CConfig& config,
+                          size_t varBegin,
+                          size_t varEnd,
+                          const FieldType& field,
+                          const GradientType& gradient,
+                          FieldType& fieldMin,
+                          FieldType& fieldMax,
+                          FieldType& limiter)
+{
+  constexpr size_t MAXNDIM = 3;
+  constexpr size_t MAXNVAR = 8;
+
+  if (varEnd > MAXNVAR)
+    SU2_MPI::Error("Number of variables is too large, increase MAXNVAR.", CURRENT_FUNCTION);
+
+  CLimiterDetails<LimiterKind> details;
```

Comment:
my only gripe is the vague name `details`

maybe `limiterDetails`

I'm really nitpicking now

Reply:
Fair enough, I'll change that.